mcp_server/jira_tools.py

```diff
@@ -398,3 +406,40 @@ async def verify_issue_author(
             group.get("name") == RH_EMPLOYEE_GROUP
             for group in user_data.get("groups", {}).get("items", [])
         )
+
+async def check_z_stream_errata_shipped(
+    issue_key: Annotated[str, Field(description="Jira issue key (e.g. RHEL-12345)")],
+    branch: Annotated[str, Field(description="Branch name (e.g. 'rhel-9.8')")],
+) -> bool:
+    """
+    Checks if the issue is for the Z-stream, if so, it checks if the issue is closed
+    (it means the errata has been shipped). If the issue is not for the Z-stream, it returns True.
+    """
+    if not re.match(r"^rhel-\d+\.\d+$", branch.lower()):
+        return True
+
+    headers = _get_jira_headers(os.getenv("JIRA_TOKEN"))
+    async with aiohttp.ClientSession() as session:
+        try:
+            async with session.get(
+                urljoin(os.getenv("JIRA_URL"), f"rest/api/2/issue/{issue_key}"),
+                params={"fields": "status, assignee"},
+                headers=headers,
+            ) as response:
+                response.raise_for_status()
+                current_issue = await response.json()
+                current_status = current_issue.get("fields", {}).get("status", {}).get("name", "")
+                current_assignee = current_issue.get("fields", {}).get("assignee", {}).get("name", "")
+        except aiohttp.ClientError as e:
+            raise ToolError(f"Failed to get Jira data: {e}") from e
+
+    # If the issue is not assigned to the Jötnar project,
+    # don't wait for it
+    # @TODO: is this ok??? for example https://issues.redhat.com/browse/RHEL-110692
+    # will wait on https://issues.redhat.com/browse/RHEL-110689
+    # without this check.
+    if current_assignee.lower() != "jötnar project":
+        return True
```

Comment:
I think we should always wait, but we can discuss this more. I would also assume this shouldn't happen most of the time? And in the given example, the second issue is not z-stream one, so it wouldn't wait, or am I missing something?